### DISCUSSION TITLE
run osrm backend with localhost

### DISCUSSION THREAD

**Discussion by giangnguyenpzh** [Project-OSRM/osrm-backend#6812] [Q&A]
**How can i run osrm backend project under my windows localhost machine?**

**Comment by washaqq** [comment]
discussed here; https://github.com/Project-OSRM/osrm-backend/issues/4941 ,
 i think you have 3 options, first download windows binaries as mentioned link above, second docker, and other option is WSL .

**Reply by jeanbaptisteb** [reply]
> discussed here; #4941 , i think you have 3 options, first download windows binaries as mentioned link above, second docker, and other option is WSL .

From experience: 
- Windows binaries are quite old, and miss features from recent versions of OSRM.
- Docker is not available to many Windows users, due to Docker's Terms of Use. 
- WSL and WSL2: I have wasted many hours on trying this solution, in vain. I've yet to find a tutorial explaining how to use OSRM with WSL. The only tutorials I've found involve using WSL with Docker, which is not a workable solution for many people, per the previous point.

The only option I've been able to run on Windows is **running OSRM with a Virtual Box** with Windows 10 as host and Ubuntu as guest. More details about the setup here: https://github.com/Project-OSRM/osrm-backend/issues/6135#issuecomment-1257652549 This is first option I'd recommend to people stuck with Windows and who can't use Docker.

**Reply by OrangeDog** [reply]
> I've yet to find a tutorial explaining how to use OSRM with WSL

Simply start a WSL shell and then follow the normal instructions. You can install docker and use the docker image, download the linux binaries, or build from source.
